### PR TITLE
[jsk_unitree_robot/cross] Fixed build for Mac environment

### PR DESCRIPTION
**Pull request by iory**
# What is this?

This is a change to do a cross compilation in the m1 mac environment.
To build it, some python libraries could not be installed on Mac.

@tkmtnt7000
Could you try this patch on your M1 Mac?
If you have any problems, please ask me.
Check if you can run `make system` and `make user`.

### CODE REVIEW

**Comment by tkmtnt7000** [comment]
@iory 
Thank you for debugging.
I'll test it on my macbook.

File: jsk_unitree_robot/cross/README.md [hunk 1]
```diff
@@ -19,6 +19,11 @@ $ newgrp
 $ sudo apt install -y qemu-user-static
 ```
 
+For Mac Users, you need to install `python-vcstool`
+```
+$ brew install python-vcstool
+```
+
```
Comment (tkmtnt7000) [inline jsk_unitree_robot/cross/README.md:26]: I cannot execute `brew install python-vcstool` in my environment.
```
$ brew install python-vcstool
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> New Formulae
install-peerdeps                                                    redis@6.2                                                           zx

==> Tapping homebrew/cask
Cloning into '/opt/homebrew/Library/Taps/homebrew/homebrew-cask'...
remote: Enumerating objects: 647014, done.
remote: Counting objects: 100% (46/46), done.
remote: Compressing objects: 100% (34/34), done.
remote: Total 647014 (delta 26), reused 25 (delta 12), pack-reused 646968
Receiving objects: 100% (647014/647014), 306.06 MiB | 13.19 MiB/s, done.
Resolving deltas: 100% (458232/458232), done.
Tapped 4014 casks (4,086 files, 326.7MB).
Warning: No available formula with the name "python-vcstool". Did you mean python-yq?
==> Searching for similarly named formulae...
This similarly named formula was found:
python-yq
To install it, run:
  brew install python-yq
==> Searching for a previously deleted formula (in the last month)...
Error: No previously deleted formula found.
==> Searching taps on GitHub...
Error: No formulae found in taps.
```
My environment
```
 base  ~  ros  jsk_robot  jsk_unitree_robot  cross  unitree  $  
$ brew --version
Homebrew 3.5.2
Homebrew/homebrew-core (git revision 50275198c04; last commit 2022-06-15)
Homebrew/homebrew-cask (git revision 4b9512207c; last commit 2022-06-15)

$ which brew
/opt/homebrew/bin/brew
```
Maybe `pip install vcstool`? or something else?

File: jsk_unitree_robot/cross/build_ros1.sh [hunk 2]
```diff
@@ -32,7 +32,15 @@ mkdir -p ${HOST_INSTALL_ROOT}/ros1_inst
 if [ ${UPDATE_SOURCE_ROOT} -eq 1 ]; then
     vcs import --force --retry 3 --shallow ${SOURCE_ROOT}/src < repos/roseus_no_window.repos
     for dir in euslisp jskeus; do ls ${SOURCE_ROOT}/src/$dir/patches/; rsync -avz ${SOURCE_ROOT}/src/$dir/patches/ ${SOURCE_ROOT}/src/$dir; done
-    sed -i s@:{version}@0.0.0@ ${SOURCE_ROOT}/src/euslisp/package.xml ${SOURCE_ROOT}/src/jskeus/package.xml
+    case ${OSTYPE} in
+        linux*)
+            sed -i s@:{version}@0.0.0@ ${SOURCE_ROOT}/src/euslisp/package.xml ${SOURCE_ROOT}/src/jskeus/package.xml
```
Comment (k-okada) [inline jsk_unitree_robot/cross/build_ros1.sh:37]: linux can use `sed -i'.bak' ` and it seems that latest mac also supports same syntax
https://stackoverflow.com/a/14813278

File: jsk_unitree_robot/cross/build_user.sh [hunk 3]
```diff
@@ -38,25 +38,58 @@ done
...
+                --cmake-args -DCATKIN_ENABLE_TESTING=FALSE \
+            " 2>&1 | tee ${TARGET_MACHINE}_build_user.log
+        ;;
+    darwin*)
+        # future module's Queue is odd.
+        # >>> from Queue import Queue
+        # ImportError: cannot import name Queue
+        # >>> import Queue
+        # >>> Queue.__file__
+        # '/opt/jsk/System/ros1_dependencies/lib/python2.7/site-packages/future-0.15.2-py2.7.egg/Queue/__init__.pyc'
```
Comment (k-okada) [inline jsk_unitree_robot/cross/build_user.sh:71]: Did you check if this does not happen in linux?

File: jsk_unitree_robot/cross/README.md [hunk 4]
```diff
@@ -19,6 +19,11 @@ $ newgrp
 $ sudo apt install -y qemu-user-static
 ```
 
+For Mac Users, you need to install `python-vcstool`
```
Comment (tkmtnt7000) [inline jsk_unitree_robot/cross/README.md:22]: For Mac Users, need to install wget
```
brew install wget
```

**Comment by tkmtnt7000** [comment]
I can run `make system` and `make user` and build successfully.

**Comment by tkmtnt7000** [comment]
Now `build_ros1_dependencies.sh` fails with following error on Mac.
```
...
[ 96%] Built target portaudio_static
[100%] Linking C shared library libportaudio.so
/usr/bin/ld: cannot find -ljack
collect2: error: ld returned 1 exit status
CMakeFiles/portaudio.dir/build.make:460: recipe for target 'libportaudio.so' failed
make[2]: *** [libportaudio.so] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/portaudio.dir/all' failed
make[1]: *** [CMakeFiles/portaudio.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
+ exit 1

```

**Comment by tkmtnt7000** [comment]
Now on linux, `./build_ros1.sh` hangs and fails.
(Others works well.)
```bash
...
==> Processing catkin package: 'pr2eus'                                                       
==> Creating build directory: 'build_isolated/pr2eus'                                         
==> Building with env: '/opt/jsk/System/ros1_inst/env.sh'                                     
==> cmake /home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus -DCATKIN_DEVEL_PREFIX=/home/user/
arm64v8_ws_system/devel_isolated/pr2eus -DCMAKE_INSTALL_PREFIX=/opt/jsk/System/ros1_inst -DCMA
KE_BUILD_TYPE=Release -DCATKIN_ENABLE_TESTING=FALSE -DEUSLISP_WITHOUT_DISPLAY=TRUE -DDISABLE_D
OCUMENTATION=1 -G Unix Makefiles in '/home/user/arm64v8_ws_system/build_isolated/pr2eus'      
-- The C compiler identification is GNU 7.5.0                                                 
-- The CXX compiler identification is GNU 7.5.0                                               
-- Check for working C compiler: /usr/bin/cc                                                  
-- Check for working C compiler: /usr/bin/cc -- works                                         
-- Detecting C compiler ABI info                                                              
-- Detecting C compiler ABI info - done                                                       
-- Detecting C compile features                                                               
-- Detecting C compile features - done                                                        
-- Check for working CXX compiler: /usr/bin/c++                                               
-- Check for working CXX compiler: /usr/bin/c++ -- works                                      
-- Detecting CXX compiler ABI info                                                            
-- Detecting CXX compiler ABI info - done                                                     
-- Detecting CXX compile features                                                             
-- Detecting CXX compile features - done                                                      
-- Using CATKIN_DEVEL_PREFIX: /home/user/arm64v8_ws_system/devel_isolated/pr2eus              
-- Using CMAKE_PREFIX_PATH: /opt/jsk/System/ros1_inst;/opt/ros/melodic;/opt/jsk/System/ros1_de
pendencies                                                                                    
-- This workspace overlays: /opt/jsk/System/ros1_inst;/opt/ros/melodic                        
-- Found PythonInterp: /usr/bin/python2 (found suitable version "2.7.17", minimum required is 
"2")                                                                                          
-- Using PYTHON_EXECUTABLE: /usr/bin/python2                                                  
-- Using Debian Python package layout                                                         
-- Using empy: /usr/bin/empy                                                                  
-- Using CATKIN_ENABLE_TESTING: FALSE                                                         
-- catkin 0.7.29                                                                              
-- BUILD_SHARED_LIBS is on                                                                    
-- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy                      
-- pr2_controllers_msgs FOUND                                                                 
-- pr2_mechanism_msgs FOUND                                                                   
-- set EUSDIR to /opt/jsk/System/ros1_inst/share/euslisp/jskeus/eus                           
-- set ARCHDIR to LinuxARM                                                                    
-- [roseus.cmake] loading roseus.cmake from pr2eus                                            
-- [roseus.cmake] roseus 1.7.5                                                                
-- [roseus.cmake] jskeus 0.0.0                                                                
-- [roseus.cmake] euslisp 0.0.0                                                               
-- [roseus.cmake] geneus 2.2.6                                                                
-- [roseus.cmake] genmsg 0.5.16                                                               
-- [roseus.cmake] message_generation Requires: gencpp;geneus;gennodejs;genlisp;genmsg;genpy   
running ROS_PACKAGE_PATH=/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus::/opt/jsk/System/r
os1_inst/share:/opt/ros/melodic/share CMAKE_PREFIX_PATH=/home/user/arm64v8_ws_system/devel_iso
lated/pr2eus:/home/user/arm64v8_ws_system/devel_isolated/pr2eus:/opt/jsk/System/ros1_inst:/opt
/ros/melodic:/opt/jsk/System/ros1_dependencies /opt/jsk/System/ros1_inst/bin/roseus lib/llib/d
ocumentation.l "(setq;lisp::*error-handler*;'exit)";"(load;\"/home/user/arm64v8_ws_system/src/
jsk_pr2eus/pr2eus/robot-interface.l\")";"(make-document;\"/home/user/arm64v8_ws_system/src/jsk
_pr2eus/pr2eus/robot-interface.l\";\"robot-interface.md\";\"\")";"(exit)"
running ROS_PACKAGE_PATH=/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus::/opt/jsk/System/r
os1_inst/share:/opt/ros/melodic/share CMAKE_PREFIX_PATH=/home/user/arm64v8_ws_system/devel_iso
lated/pr2eus:/home/user/arm64v8_ws_system/devel_isolated/pr2eus:/opt/jsk/System/ros1_inst:/opt
/ros/melodic:/opt/jsk/System/ros1_dependencies /opt/jsk/System/ros1_inst/bin/roseus lib/llib/d
ocumentation.l "(setq;lisp::*error-handler*;'exit)";"(load;\"/home/user/arm64v8_ws_system/src/
jsk_pr2eus/pr2eus/pr2-interface.l\")";"(make-document;\"/home/user/arm64v8_ws_system/src/jsk_p
r2eus/pr2eus/pr2-interface.l\";\"pr2-interface.md\";\"\")";"(exit)"
running ROS_PACKAGE_PATH=/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus::/opt/jsk/System/r
os1_inst/share:/opt/ros/melodic/share CMAKE_PREFIX_PATH=/home/user/arm64v8_ws_system/devel_iso
lated/pr2eus:/home/user/arm64v8_ws_system/devel_isolated/pr2eus:/opt/jsk/System/ros1_inst:/opt
/ros/melodic:/opt/jsk/System/ros1_dependencies /opt/jsk/System/ros1_inst/bin/roseus lib/llib/d
ocumentation.l "(setq;lisp::*error-handler*;'exit)";"(load;\"/home/user/arm64v8_ws_system/src/
jsk_pr2eus/pr2eus/pr2-utils.l\")";"(make-document;\"/home/user/arm64v8_ws_system/src/jsk_pr2eu
s/pr2eus/pr2-utils.l\";\"pr2-utils.md\";\"\")";"(exit)"
running ROS_PACKAGE_PATH=/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus::/opt/jsk/System/r
os1_inst/share:/opt/ros/melodic/share CMAKE_PREFIX_PATH=/home/user/arm64v8_ws_system/devel_iso
lated/pr2eus:/home/user/arm64v8_ws_system/devel_isolated/pr2eus:/opt/jsk/System/ros1_inst:/opt
/ros/melodic:/opt/jsk/System/ros1_dependencies /opt/jsk/System/ros1_inst/bin/roseus lib/llib/d
ocumentation.l "(setq;lisp::*error-handler*;'exit)";"(load;\"/home/user/arm64v8_ws_system/src/
jsk_pr2eus/pr2eus/speak.l\")";"(make-document;\"/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr
2eus/speak.l\";\"speak.md\";\"\")";"(exit)"
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    DISABLE_DOCUMENTATION
    EUSLISP_WITHOUT_DISPLAY

-- Build files have been written to: /home/user/arm64v8_ws_system/build_isolated/pr2eus
==> make -j12 -l12 in '/home/user/arm64v8_ws_system/build_isolated/pr2eus'
Scanning dependencies of target install_roseus
[  0%] Built target install_roseus                      
Scanning dependencies of target pr2eus_robot-interface_generate_eusdoc
Scanning dependencies of target pr2eus_pr2-utils_generate_eusdoc
Scanning dependencies of target pr2eus_pr2-interface_generate_eusdoc
Scanning dependencies of target pr2eus_speak_generate_eusdoc
[ 25%] Generating robot-interface.md
[ 50%] Generating pr2-utils.md                                                                [ 75%] Generating speak.md                                                                    [100%] Generating pr2-interface.md                                                            configuring by "/opt/jsk/System/ros1_inst/share/euslisp/jskeus/eus//lib/eusrt.l"              ;; readmacro ;; object ;; packsym ;; common ;; constants ;; stream ;; string ;; loader ;; ppri
nt ;; process ;; hashtab ;; array ;; mathtran ;; eusdebug ;; eusforeign ;; extnum ;; coordinat
es ;; tty ;; history ;; toplevel ;; trans ;; comp ;; builtins ;; l/par.l ;; intersection ;; ge
oclasses ;; geopack ;; geobody ;; primt ;; compose ;; polygon ;; viewing ;; viewport ;; viewsu
rface ;; hid ;; shadow ;; bodyrel ;; dda ;; helpsub ;; eushelp ;; pixword ;; RGBHLS ;; convolv
e ;; piximage ;; pbmfile ;; image_correlation ;; fstringdouble irtmath irtutil irtc irtgeoc ir
tgraph gnuplotlib ___time ___pgsql irtgeo euspqp pqp irtscene irtmodel irtdyna irtrobot irtsen
sor irtbvh irtcollada irtstl irtwrl irtpointcloud eusbullet bullet irtcollision eusjpeg euspng
 png irtimage
EusLisp 9.29(56fa6dd 0.0.0) for Linux created on b7b26594b3a1(Fri Jun 17 07:14:09 UTC 2022)
roseus ;; loading roseus("1.7.5") on euslisp((9.29 b7b26594b3a1 Fri Jun 17 07:14:09 UTC 2022 5
6fa6dd 0.0.0))                                                          
;; loading roseus("1.7.5") on euslisp((9.29 b7b26594b3a1 Fri Jun 17 07:14:09 UTC 2022 56fa6dd 
0.0.0))                                                                                       ;; loading roseus("1.7.5") on euslisp((9.29 b7b26594b3a1 Fri Jun 17 07:14:09 UTC 2022 56fa6dd 0.0.0))                                                                                       ;; loading roseus("1.7.5") on euslisp((9.29 b7b26594b3a1 Fri Jun 17 07:14:09 UTC 2022 56fa6dd 0.0.0))                                                                     
eustf                                
;; extending gcstack 0x5506cd3020[16374] --> 0x55008b38f0[32748] top=3f04
roseus_c_util ;; Segmentation Fault.
;; in (when ros::*compile-message* (in-package "ROS") (let* ((fname (pathname-name ros::dirfil
e)) (ros::old-module (find (format nil "~A_~A" ros::pkg fname) *loaded-modules* :key #'lisp::l
oad-module-file-name :test #'equal))) (unless ros::old-module (compiler:compile-file-if-src-ne
wer ros::dirfile (format nil "~A/~A_~A" dir ros::pkg fname)) (load (format nil "~A/~A_~A.so" d
ir ros::pkg fname) :entry (format nil "___~A" fname)))) (in-package "USER"))
;; You are still in a signal handler.
;;Try reset or throw to upper level as soon as possible.
;; code=12544800 x=be9d70 addr=5502b1da88           
signal=11 to thread 0,                                  
;; No, you cannot continue the previous evaluation.
;; Segmentation Fault.
;; in (when ros::*compile-message* (in-package "ROS") (let* ((fname (pathname-name ros::dirfile)) (ros::old-module (find (format nil "~A_~A" ros::pkg fname) *loaded-modules* :key #'lisp::load-module-file-name :test #'equal))) (unless ros::old-module (compiler:compile-file-if-src-newer ros::dirfile (format nil "~A/~A_~A" dir ros::pkg fname)) (load (format nil "~A/~A_~A.so" dir ros::pkg fnfree(): double free detected in tcache 2
ame) :entry (format nil "___~A" fname)))) (in-package "USER"))
;; You are still in a signal handler.
;;Try reset or throw to upper level as soon as possible.
;; code=14638168 x=de9430 addr=5502c2db78
signal=11 to thread 0,
;; No, you cannot continue the previous evaluation.
free(): double free detected in tcache 2
;; Segmentation Fault.
;; in (when ros::*compile-message* (in-package "ROS") (let* ((fname (pathname-name ros::dirfile)) (ros::old-module (find (format nil "~A_~A" ros::pkg fname) *loaded-modules* :key #'lisp::load-module-file-name :test #'equal))) (unless ros::old-module (compiler:compile-file-if-src-newer ros::dirfile (format nil "~A/~A_~A" dir ros::pkg fname)) (load (format nil "~A/~A_~A.so" dir ros::pkg fname) :entry (format nil "___~A" fname)))) (in-package "USER"))
;; You are still in a signal handler.
;;Try reset or throw to upper level as soon as possible.
;; code=14638176 x=de9430 addr=5502c2db78
signal=11 to thread 0,
;; No, you cannot continue the previous evaluation.

;; extending gcstack 0x55008b38f0[32738] --> 0x5500a6a370[65476] top=0

;; extending gcstack 0x5500a6a370[65466] --> 0x5500aea1b0[130932] top=0

;; extending gcstack 0x5500aea1b0[130922] --> 0x5500be9d70[261844] top=0
;; Segmentation Fault.
;; in (when ros::*compile-message* (in-package "ROS") (let* ((fname (pathname-name ros::dirfile)) (ros::old-module (find (format nil "~A_~A" ros::pkg fname) *loaded-modules* :key #'lisp::load-module-file-name :test #'equal))) (unless ros::old-module (compiler:compile-file-if-src-newer ros::dirfile (format nil "~A/~A_~A" dir ros::pkg fname)) (load (format nil "~A/~A_~A.so" dir ros::pkg fname) :entry (format nil "___~A" fname)))) (in-package "USER"))
;; You are still in a signal handler.
;;Try reset or throw to upper level as soon as possible.
;; code=12544840 x=be9d70 addr=0
Fatal: free(): double free detected in tcache 2
Aborted (core dumped)
Failed to generate robot-interface.l, but do not raise error
Aborted (core dumped)
Failed to generate pr2-utils.l, but do not raise error
Aborted (core dumped)
Failed to generate speak.l, but do not raise error
[100%] Built target pr2eus_pr2-utils_generate_eusdoc
[100%] Built target pr2eus_robot-interface_generate_eusdoc
[100%] Built target pr2eus_speak_generate_eusdoc
```

**Comment by tkmtnt7000** [comment]
> Now on linux, `./build_ros1.sh` hangs and fails. (Others works well.)
> 
> ```shell
> ...
> ==> Processing catkin package: 'pr2eus'                                                       
> ==> Creating build directory: 'build_isolated/pr2eus'                                         
> ==> Building with env: '/opt/jsk/System/ros1_inst/env.sh'                                     
> ==> cmake /home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus -DCATKIN_DEVEL_PREFIX=/home/user/
> arm64v8_ws_system/devel_isolated/pr2eus -DCMAKE_INSTALL_PREFIX=/opt/jsk/System/ros1_inst -DCMA
> KE_BUILD_TYPE=Release -DCATKIN_ENABLE_TESTING=FALSE -DEUSLISP_WITHOUT_DISPLAY=TRUE -DDISABLE_D
> OCUMENTATION=1 -G Unix Makefiles in '/home/user/arm64v8_ws_system/build_isolated/pr2eus'      
> -- The C compiler identification is GNU 7.5.0                                                 
> -- The CXX compiler identification is GNU 7.5.0                                               
> -- Check for working C compiler: /usr/bin/cc                                                  
> -- Check for working C compiler: /usr/bin/cc -- works                                         
> -- Detecting C compiler ABI info                                                              
> -- Detecting C compiler ABI info - done                                                       
> -- Detecting C compile features                                                               
> -- Detecting C compile features - done                                                        
> -- Check for working CXX compiler: /usr/bin/c++                                               
> -- Check for working CXX compiler: /usr/bin/c++ -- works                                      
> -- Detecting CXX compiler ABI info                                                            
> -- Detecting CXX compiler ABI info - done                                                     
> -- Detecting CXX compile features                                                             
> -- Detecting CXX compile features - done                                                      
> -- Using CATKIN_DEVEL_PREFIX: /home/user/arm64v8_ws_system/devel_isolated/pr2eus              
> -- Using CMAKE_PREFIX_PATH: /opt/jsk/System/ros1_inst;/opt/ros/melodic;/opt/jsk/System/ros1_de
> pendencies                                                                                    
> -- This workspace overlays: /opt/jsk/System/ros1_inst;/opt/ros/melodic                        
> -- Found PythonInterp: /usr/bin/python2 (found suitable version "2.7.17", minimum required is 
> "2")                                                                                          
> -- Using PYTHON_EXECUTABLE: /usr/bin/python2                                                  
> -- Using Debian Python package layout                                                         
> -- Using empy: /usr/bin/empy                                                                  
> -- Using CATKIN_ENABLE_TESTING: FALSE                                                         
> -- catkin 0.7.29                                                                              
> -- BUILD_SHARED_LIBS is on                                                                    
> -- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy                      
> -- pr2_controllers_msgs FOUND                                                                 
> -- pr2_mechanism_msgs FOUND                                                                   
> -- set EUSDIR to /opt/jsk/System/ros1_inst/share/euslisp/jskeus/eus                           
> -- set ARCHDIR to LinuxARM                                                                    
> -- [roseus.cmake] loading roseus.cmake from pr2eus                                            
> -- [roseus.cmake] roseus 1.7.5                                                                
> -- [roseus.cmake] jskeus 0.0.0                                                                
> -- [roseus.cmake] euslisp 0.0.0                                                               
> -- [roseus.cmake] geneus 2.2.6                                                                
> -- [roseus.cmake] genmsg 0.5.16                                                               
> -- [roseus.cmake] message_generation Requires: gencpp;geneus;gennodejs;genlisp;genmsg;genpy   
> running ROS_PACKAGE_PATH=/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus::/opt/jsk/System/r
> os1_inst/share:/opt/ros/melodic/share CMAKE_PREFIX_PATH=/home/user/arm64v8_ws_system/devel_iso
> lated/pr2eus:/home/user/arm64v8_ws_system/devel_isolated/pr2eus:/opt/jsk/System/ros1_inst:/opt
> /ros/melodic:/opt/jsk/System/ros1_dependencies /opt/jsk/System/ros1_inst/bin/roseus lib/llib/d
> ocumentation.l "(setq;lisp::*error-handler*;'exit)";"(load;\"/home/user/arm64v8_ws_system/src/
> jsk_pr2eus/pr2eus/robot-interface.l\")";"(make-document;\"/home/user/arm64v8_ws_system/src/jsk
> _pr2eus/pr2eus/robot-interface.l\";\"robot-interface.md\";\"\")";"(exit)"
> running ROS_PACKAGE_PATH=/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus::/opt/jsk/System/r
> os1_inst/share:/opt/ros/melodic/share CMAKE_PREFIX_PATH=/home/user/arm64v8_ws_system/devel_iso
> lated/pr2eus:/home/user/arm64v8_ws_system/devel_isolated/pr2eus:/opt/jsk/System/ros1_inst:/opt
> /ros/melodic:/opt/jsk/System/ros1_dependencies /opt/jsk/System/ros1_inst/bin/roseus lib/llib/d
> ocumentation.l "(setq;lisp::*error-handler*;'exit)";"(load;\"/home/user/arm64v8_ws_system/src/
> jsk_pr2eus/pr2eus/pr2-interface.l\")";"(make-document;\"/home/user/arm64v8_ws_system/src/jsk_p
> r2eus/pr2eus/pr2-interface.l\";\"pr2-interface.md\";\"\")";"(exit)"
> running ROS_PACKAGE_PATH=/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus::/opt/jsk/System/r
> os1_inst/share:/opt/ros/melodic/share CMAKE_PREFIX_PATH=/home/user/arm64v8_ws_system/devel_iso
> lated/pr2eus:/home/user/arm64v8_ws_system/devel_isolated/pr2eus:/opt/jsk/System/ros1_inst:/opt
> /ros/melodic:/opt/jsk/System/ros1_dependencies /opt/jsk/System/ros1_inst/bin/roseus lib/llib/d
> ocumentation.l "(setq;lisp::*error-handler*;'exit)";"(load;\"/home/user/arm64v8_ws_system/src/
> jsk_pr2eus/pr2eus/pr2-utils.l\")";"(make-document;\"/home/user/arm64v8_ws_system/src/jsk_pr2eu
> s/pr2eus/pr2-utils.l\";\"pr2-utils.md\";\"\")";"(exit)"
> running ROS_PACKAGE_PATH=/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr2eus::/opt/jsk/System/r
> os1_inst/share:/opt/ros/melodic/share CMAKE_PREFIX_PATH=/home/user/arm64v8_ws_system/devel_iso
> lated/pr2eus:/home/user/arm64v8_ws_system/devel_isolated/pr2eus:/opt/jsk/System/ros1_inst:/opt
> /ros/melodic:/opt/jsk/System/ros1_dependencies /opt/jsk/System/ros1_inst/bin/roseus lib/llib/d
> ocumentation.l "(setq;lisp::*error-handler*;'exit)";"(load;\"/home/user/arm64v8_ws_system/src/
> jsk_pr2eus/pr2eus/speak.l\")";"(make-document;\"/home/user/arm64v8_ws_system/src/jsk_pr2eus/pr
> 2eus/speak.l\";\"speak.md\";\"\")";"(exit)"
> -- Configuring done
> -- Generating done
> CMake Warning:
>   Manually-specified variables were not used by the project:
> 
>     DISABLE_DOCUMENTATION
>     EUSLISP_WITHOUT_DISPLAY
> 
> -- Build files have been written to: /home/user/arm64v8_ws_system/build_isolated/pr2eus
> ==> make -j12 -l12 in '/home/user/arm64v8_ws_system/build_isolated/pr2eus'
> Scanning dependencies of target install_roseus
> [  0%] Built target install_roseus                      
> Scanning dependencies of target pr2eus_robot-interface_generate_eusdoc
> Scanning dependencies of target pr2eus_pr2-utils_generate_eusdoc
> Scanning dependencies of target pr2eus_pr2-interface_generate_eusdoc
> Scanning dependencies of target pr2eus_speak_generate_eusdoc
> [ 25%] Generating robot-interface.md
> [ 50%] Generating pr2-utils.md                                                                [ 75%] Generating speak.md                                                                    [100%] Generating pr2-interface.md                                                            configuring by "/opt/jsk/System/ros1_inst/share/euslisp/jskeus/eus//lib/eusrt.l"              ;; readmacro ;; object ;; packsym ;; common ;; constants ;; stream ;; string ;; loader ;; ppri
> nt ;; process ;; hashtab ;; array ;; mathtran ;; eusdebug ;; eusforeign ;; extnum ;; coordinat
> es ;; tty ;; history ;; toplevel ;; trans ;; comp ;; builtins ;; l/par.l ;; intersection ;; ge
> oclasses ;; geopack ;; geobody ;; primt ;; compose ;; polygon ;; viewing ;; viewport ;; viewsu
> rface ;; hid ;; shadow ;; bodyrel ;; dda ;; helpsub ;; eushelp ;; pixword ;; RGBHLS ;; convolv
> e ;; piximage ;; pbmfile ;; image_correlation ;; fstringdouble irtmath irtutil irtc irtgeoc ir
> tgraph gnuplotlib ___time ___pgsql irtgeo euspqp pqp irtscene irtmodel irtdyna irtrobot irtsen
> sor irtbvh irtcollada irtstl irtwrl irtpointcloud eusbullet bullet irtcollision eusjpeg euspng
>  png irtimage
> EusLisp 9.29(56fa6dd 0.0.0) for Linux created on b7b26594b3a1(Fri Jun 17 07:14:09 UTC 2022)
> roseus ;; loading roseus("1.7.5") on euslisp((9.29 b7b26594b3a1 Fri Jun 17 07:14:09 UTC 2022 5
> 6fa6dd 0.0.0))                                                          
> ;; loading roseus("1.7.5") on euslisp((9.29 b7b26594b3a1 Fri Jun 17 07:14:09 UTC 2022 56fa6dd 
> 0.0.0))                                                                                       ;; loading roseus("1.7.5") on euslisp((9.29 b7b26594b3a1 Fri Jun 17 07:14:09 UTC 2022 56fa6dd 0.0.0))                                                                                       ;; loading roseus("1.7.5") on euslisp((9.29 b7b26594b3a1 Fri Jun 17 07:14:09 UTC 2022 56fa6dd 0.0.0))                                                                     
> eustf                                
> ;; extending gcstack 0x5506cd3020[16374] --> 0x55008b38f0[32748] top=3f04
> roseus_c_util ;; Segmentation Fault.
> ;; in (when ros::*compile-message* (in-package "ROS") (let* ((fname (pathname-name ros::dirfil
> e)) (ros::old-module (find (format nil "~A_~A" ros::pkg fname) *loaded-modules* :key #'lisp::l
> oad-module-file-name :test #'equal))) (unless ros::old-module (compiler:compile-file-if-src-ne
> wer ros::dirfile (format nil "~A/~A_~A" dir ros::pkg fname)) (load (format nil "~A/~A_~A.so" d
> ir ros::pkg fname) :entry (format nil "___~A" fname)))) (in-package "USER"))
> ;; You are still in a signal handler.
> ;;Try reset or throw to upper level as soon as possible.
> ;; code=12544800 x=be9d70 addr=5502b1da88           
> signal=11 to thread 0,                                  
> ;; No, you cannot continue the previous evaluation.
> ;; Segmentation Fault.
> ;; in (when ros::*compile-message* (in-package "ROS") (let* ((fname (pathname-name ros::dirfile)) (ros::old-module (find (format nil "~A_~A" ros::pkg fname) *loaded-modules* :key #'lisp::load-module-file-name :test #'equal))) (unless ros::old-module (compiler:compile-file-if-src-newer ros::dirfile (format nil "~A/~A_~A" dir ros::pkg fname)) (load (format nil "~A/~A_~A.so" dir ros::pkg fnfree(): double free detected in tcache 2
> ame) :entry (format nil "___~A" fname)))) (in-package "USER"))
> ;; You are still in a signal handler.
> ;;Try reset or throw to upper level as soon as possible.
> ;; code=14638168 x=de9430 addr=5502c2db78
> signal=11 to thread 0,
> ;; No, you cannot continue the previous evaluation.
> free(): double free detected in tcache 2
> ;; Segmentation Fault.
> ;; in (when ros::*compile-message* (in-package "ROS") (let* ((fname (pathname-name ros::dirfile)) (ros::old-module (find (format nil "~A_~A" ros::pkg fname) *loaded-modules* :key #'lisp::load-module-file-name :test #'equal))) (unless ros::old-module (compiler:compile-file-if-src-newer ros::dirfile (format nil "~A/~A_~A" dir ros::pkg fname)) (load (format nil "~A/~A_~A.so" dir ros::pkg fname) :entry (format nil "___~A" fname)))) (in-package "USER"))
> ;; You are still in a signal handler.
> ;;Try reset or throw to upper level as soon as possible.
> ;; code=14638176 x=de9430 addr=5502c2db78
> signal=11 to thread 0,
> ;; No, you cannot continue the previous evaluation.
> 
> ;; extending gcstack 0x55008b38f0[32738] --> 0x5500a6a370[65476] top=0
> 
> ;; extending gcstack 0x5500a6a370[65466] --> 0x5500aea1b0[130932] top=0
> 
> ;; extending gcstack 0x5500aea1b0[130922] --> 0x5500be9d70[261844] top=0
> ;; Segmentation Fault.
> ;; in (when ros::*compile-message* (in-package "ROS") (let* ((fname (pathname-name ros::dirfile)) (ros::old-module (find (format nil "~A_~A" ros::pkg fname) *loaded-modules* :key #'lisp::load-module-file-name :test #'equal))) (unless ros::old-module (compiler:compile-file-if-src-newer ros::dirfile (format nil "~A/~A_~A" dir ros::pkg fname)) (load (format nil "~A/~A_~A.so" dir ros::pkg fname) :entry (format nil "___~A" fname)))) (in-package "USER"))
> ;; You are still in a signal handler.
> ;;Try reset or throw to upper level as soon as possible.
> ;; code=12544840 x=be9d70 addr=0
> Fatal: free(): double free detected in tcache 2
> Aborted (core dumped)
> Failed to generate robot-interface.l, but do not raise error
> Aborted (core dumped)
> Failed to generate pr2-utils.l, but do not raise error
> Aborted (core dumped)
> Failed to generate speak.l, but do not raise error
> [100%] Built target pr2eus_pr2-utils_generate_eusdoc
> [100%] Built target pr2eus_robot-interface_generate_eusdoc
> [100%] Built target pr2eus_speak_generate_eusdoc
> ```

これは現在のunitreeブランチでも起こっていました。issueに移します。

**Comment by k-okada** [comment]
@tkmtnt7000 are we still need this?

**Pull request closed without merging**